feat: use publisher information to create organizations (#49)
easy-global-market/nifi-ngsild-ckan: README.md

> ### Input Data

Reviewer: also, I guess it is missing an update in the mapping table

The input data must be a valid NGSI-LD notification containing the entities to be persisted in the CKAN server.
Each entity must at least have a `title` attribute that will be used as the name of the resource. 

> A dataset must include a `publisher` attribute, which will be used as the name of the organization responsible for making the

Author: the entity can be of type `Organization ` (http://xmlns.com/foaf/spec/#term_Organization) or `Agent` (http://xmlns.com/foaf/spec/#term_Agent) . For the name we can : extract the last part in the id to only have `EGM` for example , provided the id always follows the same pattern. Or we get the entire entity from the relationship, insert it just like we do with datasets in the flow and take the name from a specific attribute.

> A dataset must include a `publisher` attribute, which will be used as the name of the organization responsible for making the

Reviewer: so `Organization` in this case. The extraction from the id will quickly end up with no-so-nice names (an id is a technical thing).

> A dataset must include a `publisher` attribute, which will be used as the name of the organization responsible for making the

Reviewer: and use the `title` attribute from the `Organization` entity.

> A dataset must include a `publisher` attribute, which will be used as the name of the organization responsible for making the

Reviewer: don't forget to update https://github.com/easy-global-market/ngsild-api-data-models/tree/master/dcat

> datasets and the resources available. 

Reviewer: no, it must be present as an attribute in an entity.

Author: the attribute is exctracted in the code as a flowfile attribute named specifically `publisherURL` , so we must at least say that (and also say that the flowfile attribute comes from an entity attribute)

Reviewer: as an user of the processor (and accompanying flow), I am not really interested about where you extracted the data. what's more, as we said, this extraction as a flow file attribute is temporary (until it is extracted from the flow file content in the processor)

### Input Data

The input data must be a valid NGSI-LD notification containing the entities to be persisted in the CKAN server.
Each entity must at least have a `title` attribute that will be used as the name of the resource. 

> A dataset must include a `publisher` attribute, which contains the id of the `Organization` entity responsible for making the

Reviewer: ```suggestion
A dataset must include a `publisher` attribute, which is a relationship to an `Organization` entity responsible for making the
```

A `Subscription` must be created to trigger the notifications sent when entities are created or updated.

Since there is no direct mapping between `dcat:Catalog` and an organization in CKAN, a `X-CKAN-OrganizationName` key-value pair
must be added to the `receiverInfo` in the `Subscription`. This value will be extracted in the processor as a flowfile 

> attribute and used as the name of the organization owning the resource.

Reviewer: OK, but explain where this `publisher` is coming from and how it is used
